src/main/java/org/jenkinsci/plugin/gitea/GiteaCreateSCMEvent.java

```diff
@@ -65,7 +65,9 @@ public GiteaCreateSCMEvent(@NonNull GiteaCreateEvent payload, @CheckForNull Stri
     public String descriptionFor(@NonNull SCMNavigator navigator) {
         String ref = getPayload().getRef();
         ref = ref.startsWith(Constants.R_HEADS) ? ref.substring(Constants.R_HEADS.length()) : ref;
-        return "Create event for branch " + ref + " in repository " + getPayload().getRepository().getName();
+        ref = ref.startsWith(Constants.R_TAGS) ? ref.substring(Constants.R_TAGS.length()) : ref;
+        String refType = getPayload().getRefType();
+        return "Create event for " + refType + " " + ref + " in repository " + getPayload().getRepository().getName();
```

Comment:
No need to create variable if it's used only once imho
```suggestion
        return "Create event for " + getPayload().getRefType() + " " + ref + " in repository " + getPayload().getRepository().getName();
```

Reply:
same in other places for `refType`